mantidproject/mantid
# Update Python interface to allow attribute validators

**Description of work.**

The Mantid Framework Python Interface (not the Workbench GUI) has been updated to allow for the use of attribute validators.

A unit test has been provided to test the functionality of a number of possible validators in python.

The following validators (previously implemented for algorithm parameters) have been tested:
- String List Validator (String)
- BoundedValidator (Float)
- ArrayBoundedValidator (Float)
- StringContainsValidator (String)

**To test:**
- In order to test the new functionality, one could observe the python unit test, "IFunction1DValidatorTest", provided.
- Alternatively, choose an existing *python* function that declares an attribute and add a validator (from the above list) to this 
  attribute so that:
`IFunction.declareAttribute(<string [attribute name]>, <float/string [value]>, <IValidator>)`

Resolves #33121

*This does not require release notes because this is one of a series of related PRs, release note will be added in concluding PR.*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
